packages/ibm-products/src/components/ConditionBuilder/utils/handleKeyboardEvents.js

```diff
@@ -0,0 +1,157 @@
+          } else {
+            traverseClockVise(eachElem, index, allElements, true, true);
+          }
+        });
+        e.preventDefault();
+        break;
+
+      case ' ':
+        if (isMultiSelect == 'true') {
+          if (document.activeElement.type !== 'button') {
+            //for button , enter key is click which already handled by framework, for other elements trigger click
+            document.activeElement?.click();
+          }
+          e.preventDefault();
+        }
+
+        break;
+      case 'Enter':
+        if (isMultiSelect !== 'true') {
+          if (document.activeElement.type !== 'button') {
+            //for button , enter key is click which already handled by framework, else trigger click
+            document.activeElement?.click();
+            focusThisField(e);
+          }
+        }
+
+        break;
+      case 'Escape':
+        //focus the corresponding field in which the popover is triggered
+        focusThisField(e);
+        break;
+      default:
+        break;
+    }
+  };
+
+  function getRows() {
+    return Array.from(
+      conditionBuilderRef.current.querySelectorAll('[role="row"]')
+    );
+  }
+
+  function getRowIndex(element) {
+    const rows = getRows();
+    return rows.findIndex((row) => row.contains(element));
+  }
+  function handleRowNavigation(e) {
+    const rows = getRows();
+
+    const currentRowIndex = getRowIndex(e.target);
+
+    navigateToNextRowCell(e, currentRowIndex, rows);
+  }
+  const navigateToNextRowCell = (e, currentRowIndex, rows) => {
+    //when the focussed element is a cell of the row which has only 1 cell (connector or statement) , focus the next row
+
+    let nextRowIndex = currentRowIndex;
+    if (e.key == 'ArrowUp') {
+      nextRowIndex =
+        currentRowIndex == 0 ? currentRowIndex : currentRowIndex - 1;
+    }
+    if (e.key == 'ArrowDown') {
+      nextRowIndex =
+        currentRowIndex == rows.length - 1
+          ? rows.length - 1
+          : currentRowIndex + 1;
+    }
+
+    let nextRow = rows[nextRowIndex];
+    let itemName = e.target.dataset.name;
+    nextRow?.querySelector(`[data-name="${itemName}"]`)?.focus();
+  };
+
+  const handleKeyPressForMainContent = (e) => {
+    switch (e.key) {
+      case 'ArrowRight':
+      case 'ArrowLeft':
+        conditionBuilderRef.current
+          .querySelectorAll(`[role="gridcell"] button`)
+          .forEach((eachElem, index, allElements) => {
+            if (e.key == 'ArrowRight') {
+              traverseClockVise(eachElem, index, allElements);
+            } else {
+              traverseReverse(eachElem, index, allElements);
+            }
+          });
+
+        break;
+
+      case 'ArrowUp':
+      case 'ArrowDown':
+        handleRowNavigation(e);
+
+        break;
+      case 'Enter':
+        // document.activeElement?.click();
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  let activeElement = document.activeElement;
```

Comment:
As we not reassigning we can use `const` instead of `let`.

Reply:
modified as suggested